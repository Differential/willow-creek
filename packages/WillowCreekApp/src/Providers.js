import { nest } from 'recompose';
import { Providers } from '@apollosproject/ui-kit';
import ClientProvider from './client';

export default nest(ClientProvider, Providers);
