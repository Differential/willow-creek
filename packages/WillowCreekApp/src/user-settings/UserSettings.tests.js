import React from 'react';
import renderer from 'react-test-renderer';
import wait from 'waait';

import Providers from 'WillowCreekApp/src/Providers';

import getLoginState from '../tabs/connect/getLoginState';
import UserSettings from '.';

describe('UserSettings component', () => {
  it('renders UserSettings when logged in', async () => {
    const mock = {
      request: {
        query: getLoginState,
      },
      result: {
        data: { isLoggedIn: true },
      },
    };
    const navigation = { navigate: jest.fn(), getParam: jest.fn() };
    const tree = renderer.create(
      <Providers mocks={[mock]}>
        <UserSettings navigation={navigation} />
      </Providers>
    );
    await wait(0); // wait for response from graphql
    expect(tree).toMatchSnapshot();
  });
});
