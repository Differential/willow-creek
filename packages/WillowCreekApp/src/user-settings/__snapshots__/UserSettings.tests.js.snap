// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSettings component renders UserSettings when logged in 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <BVLinearGradient
      colors={
        Array [
          4294967295,
          4294506484,
        ]
      }
      locations={null}
      style={
        Object {
          "flex": 1,
          "paddingTop": 28,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingHorizontal": 16,
            "paddingVertical": 16,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 40,
                    "height": 80,
                    "justifyContent": "flex-end",
                    "marginBottom": 8,
                    "marginRight": 0,
                    "overflow": "hidden",
                    "width": 80,
                  }
                }
                themeSize={80}
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="80"
                  bbWidth="80"
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      undefined,
                      false,
                      Object {
                        "flex": 0,
                        "height": 80,
                        "width": 80,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                >
                  <RNSVGPath
                    d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-3.33 8.33C8.67 8.5 10.2 7 12 7s3.33 1.5 3.33 3.33v.84c0 1.84-1.53 3.33-3.33 3.33S8.67 13 8.67 11.17v-.84zm3.33 10c-2.03 0-3.9-.73-5.33-1.93.7-1.32 2.07-2.23 3.66-2.23h3.34c1.6 0 2.97.9 3.66 2.22-1.44 1.2-3.3 1.92-5.33 1.92z"
                    fill={
                      Array [
                        0,
                        0.6470588235294118,
                        0.6470588235294118,
                        0.6470588235294118,
                        1,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={1}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth="1"
                    x={0}
                    y={0}
                  />
                </RNSVGSvgView>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Medium",
                "fontSize": 14,
                "lineHeight": 20.16,
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Object {
                  "color": "#17B582",
                }
              }
            >
              Change Photo
            </Text>
          </Text>
        </View>
      </View>
      <BVLinearGradient
        colors={
          Array [
            4294967295,
            4294506484,
          ]
        }
        locations={null}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderBottomWidth": 0.5,
              "borderColor": "rgba(165, 165, 165, 0.30000000000000004)",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0.5,
              "marginBottom": 16,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "flex-start",
                  "padding": 8,
                  "paddingLeft": 10.666666666666666,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#303030",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontFamily": "InterUI-Medium",
                    "fontSize": 14,
                    "lineHeight": 20.16,
                    "paddingLeft": 0,
                    "paddingRight": 8,
                  }
                }
              >
                Personal Details
              </Text>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="20"
                bbWidth="20"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                    false,
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
              >
                <RNSVGPath
                  d="M9.65 21L8 20.1l7.72-8.1L8 3.9 9.65 3l8.16 8.54c.27.28.27.64 0 .92L9.67 21z"
                  fill={
                    Array [
                      0,
                      0.18823529411764706,
                      0.18823529411764706,
                      0.18823529411764706,
                      1,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth="1"
                  x={0}
                  y={0}
                />
              </RNSVGSvgView>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "rgba(165, 165, 165, 0.30000000000000004)",
                "height": 0.5,
                "marginLeft": 10.666666666666666,
                "width": "100%",
              }
            }
          />
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "flex-start",
                  "padding": 8,
                  "paddingLeft": 10.666666666666666,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#303030",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontFamily": "InterUI-Medium",
                    "fontSize": 14,
                    "lineHeight": 20.16,
                    "paddingLeft": 0,
                    "paddingRight": 8,
                  }
                }
              >
                Change Password
              </Text>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="20"
                bbWidth="20"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                    false,
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
              >
                <RNSVGPath
                  d="M9.65 21L8 20.1l7.72-8.1L8 3.9 9.65 3l8.16 8.54c.27.28.27.64 0 .92L9.67 21z"
                  fill={
                    Array [
                      0,
                      0.18823529411764706,
                      0.18823529411764706,
                      0.18823529411764706,
                      1,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth="1"
                  x={0}
                  y={0}
                />
              </RNSVGSvgView>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderBottomWidth": 0.5,
              "borderColor": "rgba(165, 165, 165, 0.30000000000000004)",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0.5,
              "marginBottom": 16,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "flex-start",
                  "padding": 8,
                  "paddingLeft": 10.666666666666666,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#303030",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontFamily": "InterUI-Medium",
                    "fontSize": 14,
                    "lineHeight": 20.16,
                    "paddingLeft": 0,
                    "paddingRight": 8,
                  }
                }
              >
                Give Feedback
              </Text>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="20"
                bbWidth="20"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                    false,
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
              >
                <RNSVGPath
                  d="M9.65 21L8 20.1l7.72-8.1L8 3.9 9.65 3l8.16 8.54c.27.28.27.64 0 .92L9.67 21z"
                  fill={
                    Array [
                      0,
                      0.18823529411764706,
                      0.18823529411764706,
                      0.18823529411764706,
                      1,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth="1"
                  x={0}
                  y={0}
                />
              </RNSVGSvgView>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderBottomWidth": 0.5,
              "borderColor": "rgba(165, 165, 165, 0.30000000000000004)",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0.5,
              "marginBottom": 16,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "flex-start",
                  "padding": 8,
                  "paddingLeft": 10.666666666666666,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#303030",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontFamily": "InterUI-Medium",
                    "fontSize": 14,
                    "lineHeight": 20.16,
                    "paddingLeft": 0,
                    "paddingRight": 8,
                  }
                }
              >
                Privacy Policy
              </Text>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="20"
                bbWidth="20"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                    false,
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
              >
                <RNSVGPath
                  d="M9.65 21L8 20.1l7.72-8.1L8 3.9 9.65 3l8.16 8.54c.27.28.27.64 0 .92L9.67 21z"
                  fill={
                    Array [
                      0,
                      0.18823529411764706,
                      0.18823529411764706,
                      0.18823529411764706,
                      1,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth="1"
                  x={0}
                  y={0}
                />
              </RNSVGSvgView>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "rgba(165, 165, 165, 0.30000000000000004)",
                "height": 0.5,
                "marginLeft": 10.666666666666666,
                "width": "100%",
              }
            }
          />
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "flex-start",
                  "padding": 8,
                  "paddingLeft": 10.666666666666666,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#303030",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontFamily": "InterUI-Medium",
                    "fontSize": 14,
                    "lineHeight": 20.16,
                    "paddingLeft": 0,
                    "paddingRight": 8,
                  }
                }
              >
                Terms of Use
              </Text>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="20"
                bbWidth="20"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                    false,
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
              >
                <RNSVGPath
                  d="M9.65 21L8 20.1l7.72-8.1L8 3.9 9.65 3l8.16 8.54c.27.28.27.64 0 .92L9.67 21z"
                  fill={
                    Array [
                      0,
                      0.18823529411764706,
                      0.18823529411764706,
                      0.18823529411764706,
                      1,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth="1"
                  x={0}
                  y={0}
                />
              </RNSVGSvgView>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderBottomWidth": 0.5,
              "borderColor": "rgba(165, 165, 165, 0.30000000000000004)",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0.5,
              "marginBottom": 16,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "flex-start",
                  "padding": 8,
                  "paddingLeft": 10.666666666666666,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#303030",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontFamily": "InterUI-Medium",
                    "fontSize": 14,
                    "lineHeight": 20.16,
                    "paddingLeft": 0,
                    "paddingRight": 8,
                  }
                }
              >
                Logout
              </Text>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="20"
                bbWidth="20"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                    false,
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
              >
                <RNSVGPath
                  d="M9.65 21L8 20.1l7.72-8.1L8 3.9 9.65 3l8.16 8.54c.27.28.27.64 0 .92L9.67 21z"
                  fill={
                    Array [
                      0,
                      0.18823529411764706,
                      0.18823529411764706,
                      0.18823529411764706,
                      1,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth="1"
                  x={0}
                  y={0}
                />
              </RNSVGSvgView>
            </View>
          </View>
        </View>
      </BVLinearGradient>
    </BVLinearGradient>
  </View>
</RCTSafeAreaView>
`;
